x-pack/plugins/cases/public/components/all_cases/table_filter_config/use_filter_config.tsx

```diff
@@ -0,0 +1,144 @@
+    `${appId}.${LOCAL_STORAGE_KEYS.casesTableFiltersConfig}`,
+    []
+  );
+
+  useEffect(() => {
+    const newFilterConfig = mergeSystemAndCustomFieldConfigs({
+      systemFilterConfig,
+      customFieldsFilterConfig,
+    });
+
+    filterConfigs.forEach((filter) => {
+      if (!newFilterConfig.has(filter.key)) {
+        filter.deactivate();
+      }
+    });
+  }, [filterConfigs, systemFilterConfig, customFieldsFilterConfig]);
+
+  useEffect(() => {
+    setFilterConfigs(
+      mergeSystemAndCustomFieldConfigs({
+        systemFilterConfig,
+        customFieldsFilterConfig,
+      })
+    );
+  }, [systemFilterConfig, customFieldsFilterConfig]);
+
+  const onChange = ({ selectedOptionKeys }: { filterId: string; selectedOptionKeys: string[] }) => {
+    const newActiveByFilterKey = [...(activeByFilterKey || [])];
+    const deactivatedFilters: string[] = [];
+
+    // for each filter in the current state, this way we keep the order
+    (activeByFilterKey || []).forEach(({ key, isActive: prevIsActive }, currentIndex) => {
+      if (filterConfigs.has(key)) {
+        const isActive = selectedOptionKeys.find((optionKey) => optionKey === key);
+        if (isActive && !prevIsActive) {
+          // remove/insert to the end with isActive = true
+          newActiveByFilterKey.splice(currentIndex, 1);
+          newActiveByFilterKey.push({ key, isActive: true });
+        } else if (!isActive && prevIsActive) {
+          // dont move, just update isActive = false
+          deactivatedFilters.push(key);
+          newActiveByFilterKey[currentIndex] = { key, isActive: false };
+        }
+      } else {
+        // clean up filters that are no longer available
+        newActiveByFilterKey.splice(currentIndex, 1);
+      }
+    });
+
+    // for each filter in the config
+    filterConfigs.forEach(({ key: configKey }) => {
+      // add it if its a new filter
+      if (!newActiveByFilterKey.find(({ key }) => key === configKey)) {
+        // first time, the current state is empty, all filters will be added
+        // isActive = true if the filter is in the selectedOptionKeys
+        const isActive = selectedOptionKeys.find((optionKey) => optionKey === configKey);
+        if (!isActive) {
+          // for system filter that is removed as first action
+          deactivatedFilters.push(configKey);
+        }
+        newActiveByFilterKey.push({
+          key: configKey,
+          isActive: Boolean(isActive),
+        });
+      }
+    });
+
+    deactivatedFilters
+      .filter((key) => filterConfigs.has(key))
+      .forEach((key) => (filterConfigs.get(key) as FilterConfig).deactivate());
+
+    setActiveByFilterKey(newActiveByFilterKey);
+  };
+
+  const filterConfigArray = Array.from(filterConfigs.values());
+  const selectableOptions = filterConfigArray
+    .map(({ key, label }) => ({
+      key,
+      label,
+    }))
+    .sort((a, b) => {
```

Comment:
I find it a bit weird that the order in which we display the options is not the order in which the filters show

<img width="755" alt="Screenshot 2023-11-28 at 14 25 56" src="https://github.com/elastic/kibana/assets/1533137/e9b9add8-27b8-40fc-98d7-bae029905e3e">

I am not saying it is necessarily wrong, I just got accustomed to the ordering in the other popover.

My question is, are we planning to allow reordering of filters in the future?

Reply:
I'm not aware of anything that isn't reorder by deactivating/activating filters as you need it